### PR TITLE
GH-102748: remove legacy support for generator based coroutines from `asyncio.iscoroutine`

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-102748

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -237,6 +237,10 @@ asyncio
 * Add C implementation of :func:`asyncio.current_task` for 4x-6x speedup.
   (Contributed by Itamar Ostricher and Pranav Thulasiram Bhat in :gh:`100344`.)
 
+* :func:`asyncio.iscoroutine` now returns :const:`False` for generators as
+  :mod:`asyncio` does not supports legacy generator based coroutines.
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst]:
```suggestion
  :mod:`asyncio` does not supports legacy generator-based coroutines.
```

File: Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+:func:`asyncio.iscoroutine` now returns :const:`False` for generators as :mod:`asyncio` does not supports legacy generator based coroutines. Patch by Kumar Aditya.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst]: (Should probably also wrap this to 80 chars per line)

```suggestion
:func:`asyncio.iscoroutine` now returns :const:`False` for generators as :mod:`asyncio` does not supports legacy generator-based coroutines. Patch by Kumar Aditya.
```

**Review by AlexWaygood — COMMENTED** [review]
Tiny nit:

File: Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst (hunk 2 above)
Comment (kumaraditya303) [reply Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst]: Thanks committed manually

File: Doc/whatsnew/3.12.rst (hunk 1 above)
Comment (mokko) [reply Doc/whatsnew/3.12.rst]: does not support, not supports

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -237,6 +237,10 @@ asyncio
 * Add C implementation of :func:`asyncio.current_task` for 4x-6x speedup.
   (Contributed by Itamar Ostricher and Pranav Thulasiram Bhat in :gh:`100344`.)
 
+* :func:`asyncio.iscoroutine` now returns :const:`False` for generators as
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst]: As per the devguide's recommendation: https://devguide.python.org/documentation/markup/#quick-reference

The reason why is that we assume that the vast majority of Python devs reading the docs will know what True, False and None refer to, so the link will just be distracting.

```suggestion
* :func:`asyncio.iscoroutine` now returns ``False`` for generators as
```

File: Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst [hunk 4]
```diff
@@ -0,0 +1,3 @@
+:func:`asyncio.iscoroutine` now returns :const:`False` for generators as
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst]:
```suggestion
:func:`asyncio.iscoroutine` now returns ``False`` for generators as
```

**Review by AlexWaygood — COMMENTED** [review]
Another tiny docs nit

File: Doc/whatsnew/3.12.rst [hunk 5]
```diff
@@ -237,6 +237,10 @@ asyncio
...
+  :mod:`asyncio` does not supports legacy generator-based coroutines.
```
Comment (AlexWaygood) [inline Doc/whatsnew/3.12.rst]:
```suggestion
  :mod:`asyncio` does not support legacy generator-based coroutines.
```

File: Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+:mod:`asyncio` does not supports legacy generator-based coroutines.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2023-03-16-08-17-29.gh-issue-102748.WNACpI.rst]:
```suggestion
:mod:`asyncio` does not support legacy generator-based coroutines.
```

File: Doc/whatsnew/3.12.rst (hunk 3 above)
Comment (kumaraditya303) [reply Doc/whatsnew/3.12.rst]: Ok, thanks

File: Doc/whatsnew/3.12.rst (hunk 5 above)
Comment (kumaraditya303) [reply Doc/whatsnew/3.12.rst]: I trust you for the best grammar ;)

**Review by AlexWaygood — APPROVED** [review]
LGTM

File: Doc/whatsnew/3.12.rst (hunk 1 above)
Comment (AlexWaygood) [reply Doc/whatsnew/3.12.rst]: Thanks @mokko!

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**